quantum/process_keycode/process_kombo.c

```diff
@@ -0,0 +1,145 @@
+/* Copyright 2019 Bence Domonkos
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef KOMBO_ENABLE
+
+#include "process_kombo.h"
+
+extern kombo_t key_kombos[];
+uint64_t static kombo_key_state = 0;
+uint16_t static key_buffer[64];
+uint16_t static kombo_hkey = KC_NO;
+size_t static n_buffer = 0;
+bool static kombo_is_enabled = true;
+
+// should be specific for every other keyboard matrices
+// should return a number between 0-63 (inclusive) for kombo keys and -1 for the others
+extern int8_t
+kombo_get_pos(uint8_t row, uint8_t col);
+
+__attribute__ ((weak))
+bool
+process_kombo_event(size_t kombo_index, keypos_t pos) {
+	return true;
+}
+
+static void
+kombo_init(void) {
+	n_buffer = 0;
+	kombo_key_state = 0;
+	kombo_hkey = KC_NO;
+}
+
+void
+kombo_toggle(void) {
+	kombo_is_enabled ? kombo_disable() : kombo_enable();
```

Comment:
In general, we try to avoid this:
https://docs.qmk.fm/#/contributing?id=coding-conventions

Also, this may work better: 

```suggestion
  kombo_is_enabled ^= 1;
```